packages/protocol/test/mainnet_forking/blockchain_scripts/migrations/v2.4/migrate.test.ts

```diff
@@ -0,0 +1,119 @@
+
+  return {
+    gfi,
+    communityRewards,
+    communityRewardsStartingBalance,
+  }
+})
+
+describe("v2.4", async function () {
+  this.timeout(TEST_TIMEOUT)
+
+  let gfi: GFIInstance
+  let communityRewards: CommunityRewardsInstance
+  let communityRewardsStartingBalance: BN
+
+  beforeEach(async () => {
+    // eslint-disable-next-line @typescript-eslint/no-extra-semi
+    ;({gfi, communityRewards, communityRewardsStartingBalance} = await setupTest())
+    console.log("communityRewardsStartingBalance", communityRewardsStartingBalance.toString())
+  })
+
+  it("deploys BackerMerkleDirectDistributor", async () => {
+    await expect(deployments.get("BackerMerkleDirectDistributor")).to.not.be.rejected
+  })
+
+  it("initializes BackerMerkleDirectDistributor", async () => {
+    await expect(
+      (
+        await getTruffleContract<BackerMerkleDirectDistributorInstance>("BackerMerkleDirectDistributor")
+      ).initialize(ZERO_ADDRESS, ZERO_ADDRESS, web3.utils.keccak256("test"), {from: await getProtocolOwner()})
+    ).to.be.rejectedWith(/initialized/)
+  })
+
+  it("deploys BackerMerkleDistributor", async () => {
+    await expect(deployments.get("BackerMerkleDistributor")).to.not.be.rejected
+  })
+
+  expectProxyOwner({
+    toBe: async () => getProtocolOwner(),
+    forContracts: ["BackerMerkleDirectDistributor"],
+  })
+
+  expectOwnerRole({
+    toBe: async () => getProtocolOwner(),
+    forContracts: ["BackerMerkleDirectDistributor"],
+  })
+
+  expectRoles([
+    {
+      contractName: "CommunityRewards",
+      roles: [DISTRIBUTOR_ROLE],
+      address: async () => (await deployments.get("BackerMerkleDistributor")).address,
+    },
+  ])
+
+  it("loads GFI into CommunityRewards using merkleDistributorInfo amount", async () => {
+    const merkleDistributorInfo = JSON.parse(String(await fs.readFile(migrate2_4.merkleDistributorInfoPath)))
+    if (!isMerkleDistributorInfo(merkleDistributorInfo)) {
+      throw new Error("Invalid merkle distributor info")
+    }
+    const merkleDistributorAmount = web3.utils.toBN(merkleDistributorInfo.amountTotal)
+
+    const rewardsAvailable = await communityRewards.rewardsAvailable()
+
+    expect(rewardsAvailable.sub(communityRewardsStartingBalance)).to.bignumber.eq(merkleDistributorAmount)
+  })
+
+  it("loads GFI into BackerMerkleDirectDistributor using merkleDirectDistributorInfo amount", async () => {
+    const merkleDirectDistributorInfo = JSON.parse(
+      String(await fs.readFile(migrate2_4.merkleDirectDistributorInfoPath))
+    )
+    if (!isMerkleDirectDistributorInfo(merkleDirectDistributorInfo)) {
+      throw new Error("Invalid merkle direct distributor info")
+    }
+    const merkleDirectDistributorAmount = web3.utils.toBN(merkleDirectDistributorInfo.amountTotal)
+
+    const directDistributor = await deployments.get("BackerMerkleDirectDistributor")
+
+    expect(await gfi.balanceOf(directDistributor.address)).to.bignumber.eq(merkleDirectDistributorAmount)
+  })
+})
```

Comment:
also add a test that we send the payment to Igor (and/or Kotoya)

Reply:
:+1: still waiting on address